### PR TITLE
feat(commonjs): export properties defined using Object.defineProperty(exports, ..)

### PR DESCRIPTION
**Pull request by edeustace**
## Rollup Plugin Name: `commonjs `

This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### Description

This pr adds exports that have been defined using the `Object.defineProperty(exports, ...)` technique. I was running into an issue with a 3rd party lib (`@material-ui/core@3.9.4`) where it was using this technique to define the exports. Rollup was falling over with the error `export 'foo' is not exported by 'path/to/module'`.

### CODE REVIEW

File: packages/commonjs/src/transform.js [hunk 1]
```diff
@@ -299,6 +318,10 @@ export function transformCommonjs(
         return;
       }
 
+      // Is this a call to Object.defineProperty(exports, ...)?
+      const def = defineProperty(node, 'exports');
```
Comment (lukastaegert) [inline packages/commonjs/src/transform.js]: I would rename the function to `isDefinePropertyCall`. Then the comment is also not needed. The return value could be `propertyName`, then I do not have to check the function definition what it means.

Comment (edeustace) [reply packages/commonjs/src/transform.js]: done @lukastaegert :+1:

Comment (edeustace) [reply packages/commonjs/src/transform.js]: actually on reflection I propose it be called `getDefinePropertyCallName`, the java programmer in me is complaining that an `isX` function should only return a boolean. See latest commit. That  said - happy to rollback if you prefer `isDefinePropertyCall`.

Comment (shellscape) [reply packages/commonjs/src/transform.js]: Why not return a boolean?

Comment (edeustace) [reply packages/commonjs/src/transform.js]: We could, but we’d also need to get the property name to add to namedExports?

Comment (edeustace) [reply packages/commonjs/src/transform.js]: We could, but we’d also need to get the property name to add to namedExports?

Comment (lukastaegert) [reply packages/commonjs/src/transform.js]: You are right, and you definitely got what I meant 😉

Comment (shellscape) [reply packages/commonjs/src/transform.js]: Looks good. :shipit:

**Comment by shellscape** [comment]
thanks!

**Pull request merged**